src/ml/neural_net/style_transfer/mps_style_transfer_utils.m

```diff
@@ -0,0 +1,657 @@
+  NSMutableDictionary<NSString *, NSDictionary *> *vgg16Weight;
+
+  NSMutableDictionary<NSString *, NSData *> *vgg16WeightBlock1;
+  NSMutableDictionary<NSString *, NSData *> *vgg16WeightBlock2;
+  NSMutableDictionary<NSString *, NSData *> *vgg16WeightBlock3;
+  NSMutableDictionary<NSString *, NSData *> *vgg16WeightBlock4;
+
+  vgg16WeightBlock1[@"vgg_block_1block_1_conv_1_weights"] = weights[@"vgg16_conv0_weight"];
+  vgg16WeightBlock1[@"vgg_block_1block_1_conv_1_biases"] = weights[@"vgg16_conv0_bias"];
+  vgg16WeightBlock1[@"vgg_block_1block_1_conv_2_weights"] = weights[@"vgg16_conv1_weight"];
+  vgg16WeightBlock1[@"vgg_block_1block_1_conv_2_biases"] = weights[@"vgg16_conv1_bias"];
+
+  vgg16Weight[@"vgg_block_1"] = [vgg16WeightBlock1 copy];
+
+  vgg16WeightBlock2[@"vgg_block_2block_1_conv_1_weights"] = weights[@"vgg16_conv2_weight"];
+  vgg16WeightBlock2[@"vgg_block_2block_1_conv_1_biases"] = weights[@"vgg16_conv2_bias"];
+  vgg16WeightBlock2[@"vgg_block_2block_1_conv_2_weights"] = weights[@"vgg16_conv3_weight"];
+  vgg16WeightBlock2[@"vgg_block_2block_1_conv_2_biases"] = weights[@"vgg16_conv3_bias"];
+
+  vgg16Weight[@"vgg_block_2"] = [vgg16WeightBlock2 copy];
+
+  vgg16WeightBlock3[@"vgg_block_3block_2_conv_1_weights"] = weights[@"vgg16_conv4_weight"];
+  vgg16WeightBlock3[@"vgg_block_3block_2_conv_1_biases"] = weights[@"vgg16_conv4_bias"];
+  vgg16WeightBlock3[@"vgg_block_3block_2_conv_2_weights"] = weights[@"vgg16_conv5_weight"];
+  vgg16WeightBlock3[@"vgg_block_3block_2_conv_2_biases"] = weights[@"vgg16_conv5_bias"];
+  vgg16WeightBlock3[@"vgg_block_3block_2_conv_2_weights"] = weights[@"vgg16_conv6_weight"];
+  vgg16WeightBlock3[@"vgg_block_3block_2_conv_2_biases"] = weights[@"vgg16_conv6_bias"];
+
+  vgg16Weight[@"vgg_block_3"] = [vgg16WeightBlock3 copy];
+
+  vgg16WeightBlock4[@"vgg_block_4block_2_conv_1_weights"] = weights[@"vgg16_conv7_weight"];
+  vgg16WeightBlock4[@"vgg_block_4block_2_conv_1_biases"] = weights[@"vgg16_conv7_bias"];
+  vgg16WeightBlock4[@"vgg_block_4block_2_conv_2_weights"] = weights[@"vgg16_conv8_weight"];
+  vgg16WeightBlock4[@"vgg_block_4block_2_conv_2_biases"] = weights[@"vgg16_conv8_bias"];
+  vgg16WeightBlock4[@"vgg_block_4block_2_conv_2_weights"] = weights[@"vgg16_conv9_weight"];
+  vgg16WeightBlock4[@"vgg_block_4block_2_conv_2_biases"] = weights[@"vgg16_conv9_bias"];
+
+  vgg16Weight[@"vgg_block_4"] = [vgg16WeightBlock4 copy];
+
+  return [vgg16Weight copy];
+}
+
++ (void)populateMean:(NSMutableData *)data {
+  NSUInteger dataSize = (data.length)/sizeof(float);
+  NSAssert((dataSize) % 3 == 0, @"Data must follow a 3 channel format");
+  float meanWeights[3] = { 123.68, 116.779, 103.939 };
+  for(NSUInteger x = 0; x < (dataSize/3); x++){
+    for(NSUInteger c = 0;c < 3;c++){
+      NSUInteger offset = x * 3;
+      NSUInteger index = offset + c;
+      ((float *)data.mutableBytes)[index] = meanWeights[c];
+    }
+  }
```

Comment:
FWIW, consider this alternative implementation of this loop:
```
float* ptr = (float*) data.mutableBytes;
float* end = ptr + dataSize;
while (ptr < end) {
  ptr[0] = meanWeights[0];
  ptr[1] = meanWeights[1];
  ptr[2] = meanWeights[2];
  ptr += 3;
}
```

Reply:
Fixed!